compiler/src/dmd/toir.d

```diff
@@ -842,6 +842,7 @@ void buildClosure(FuncDeclaration fd, IRState *irs)
         foreach (v; fd.closureVars)
         {
             //printf("closure var %s\n", v.toChars());
+            v.inClosure = true;
```

Comment:
> 1. the backend tree doesn't exist yet in DMD when this code is run
> 
Just to select one example, a few lines down from this, there is the following 
```
            /* Set Sscope to closure */ 
             Symbol *vsym = toSymbol(v); 
             assert(vsym.Sscope == null); 
             vsym.Sscope = sclosure;
```
So why can't you set `vsym.inClosure = true;` instead? Although surely that bool flag is implied by the `Sscope` setting instead.

Having duplicate information in separate trees isn't a bad thing, but front-end representation should be manipulated in the semantic passes only.

Reply:
The symbol table for the local variables is created later.

Reply:
Why? (Though that might be an explanation for all the bugs and oddities in dmd codegen)